sql/expression_value.h

```diff
@@ -1187,7 +1193,7 @@ struct ExpressionValue {
     /** Return the most specialized possible value info for this given value.
         Used for static analysis of constants.
     */
-    std::shared_ptr<ExpressionValueInfo> getSpecializedValueInfo() const;
+    std::shared_ptr<ExpressionValueInfo> getSpecializedValueInfo(bool constant) const;
```

Comment:
It is strange that we have to pass the constness here?  Don't we already know if the `this` is const or not?